doc/user/flycheck-versus-flymake.rst

```diff
@@ -44,71 +37,68 @@ Please do **not** use this table alone to make your personal judgment.  Read the
++---------------------------+-----------------------+-----------------------+
+|Built-in                   |no [#]_                |yes                    |
++---------------------------+-----------------------+-----------------------+
+|`Enables automatically if  |yes                    |no                     |
+|possible <Enabling Syntax  |                       |                       |
+|Checking_>`_               |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Checks after               |save, newline, change  |newline, change        |
++---------------------------+-----------------------+-----------------------+
+|Checks in background       |yes                    |yes                    |
++---------------------------+-----------------------+-----------------------+
+|`Automatic syntax checker  |by major mode and      |user-defined [#]_      |
+|selection <Syntax checker  |custom predicates      |                       |
+|selection_>`_              |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Manual syntax checker     |yes                    |yes                    |
+|selection <Manual          |                       |                       |
+|Selection_>`_              |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Multiple syntax checkers  |yes                    |yes                    |
+|per buffer`_               |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Supported languages`_     |>50                    |2 [#]_                 |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Checking remote files      |said to work, but not  |partly?                |
+|via Tramp                  |officially supported   |                       |
+|                           |[#]_                   |                       |
++---------------------------+-----------------------+-----------------------+
+|`Definition of new         |single declarative     |function definition    |
+|syntax checkers`_          |function/macro         |[#]_                   |
++---------------------------+-----------------------+-----------------------+
+|`Error levels`_            |errors, warnings,      |errors, warnings,      |
+|                           |infos, custom levels   |notes, custom levels   |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Error identifiers`_       |yes                    |no                     |
```

Comment:
Don't know exactly what this...

Reply:
Flycheck has first-class support for error codes (think `eslint` lint identifiers, or `gcc` warnings).  See [here](http://www.flycheck.org/en/1403-update-flymake-comparison/user/flycheck-versus-flymake.html#error-identifiers) for how they are used.

Reply:
Hi, after a very long delay. Are you talking in this point about diagnostic sub-types? Or arbitrary error types that a backend may define.

Reply:
Here is a diagnostic from GCC:
```
foo.c:2:7: warning: unused variable ‘x’ [-Wunused-variable]
                                         ^^^^^^^^^^^^^^^^^
                                         error identifier
```
We extract the identifier along the diagnostic message, error level, filename, line and column information.  These identifiers are typically used to suppress diagnostics, either in a configuration file or inline.  See #1436 for example.

Reply:
OK @fmdkdd , I get it. Then it's safe to say that flymake doesn't have this yet (it could be implemented by a backend, but not practical).